enhancements/network/infiniband.md

```diff
@@ -0,0 +1,139 @@
+---
+title: infiniband
+authors:
+  - "@zshi"
+reviewers:
+  - "@pliurh"
+  - "@dougbtv"
+  - "@s1061123"
+  - "@danwinship"
+  - "@squeed"
+  - "@dcbw"
+  - "@yrobla"
+approvers:
+  - "@knobunc"
+  - "@fepan"
+creation-date: 2020-05-06
+last-updated: 2020-05-06
+status: implementable
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# InfiniBand
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [X] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+RDMA (Remote Direct Memory Access) fabrics such as InfiniBand and RDMA over
+Converged Ethernet (RoCE) are key to unlocking scalable performance for
+demanding Artificial Intelligence (AI) and Machine Learning (ML) applications.
+RoCE v2 is supported in SR-IOV Operator as Tech Preview since 4.3, This
+enhancement is to introduce native InfiniBand support.
+
+## Motivation
+
+Enable high-performance message transmission via RDMA for distributed workloads
+such as AI/ML.
+
+### Goals
+
+- Provide native InfiniBand support on pod additional network interfaces
+
+### Non-Goals
+
+- Support RoCE v2
+- Support GPUDirect RDMA
+- Support Ethernet over InfiniBand
+- Support IP over InfiniBand (IPoIB)
+- Support InfiniBand on pod primary network interface
+
+## Proposal
+
+Extend SR-IOV Operator to support both Ethernet and InfiniBand devices.
+
+SR-IOV Hardware NIC is managed by SR-IOV Operator and will be provisioned
+with a new device type `infiniBand` or `ib`.
+
+A new Container Network Interface (CNI) plugin `ib-sriov-cni` will be added.
+
+InfiniBand support will only be available on bare metal deployment.
+
+#### Hardware
+
+The following Mellanox NICs will be supported:
+- Mellanox Technologies MT27700 Family [ConnectX-4]
+- Mellanox Technologies MT28908 Family [ConnectX-6]
+
+#### InfiniBand SR-IOV CNI
+
+InfiniBand SR-IOV CNI `ib-sriov-cni` is a 3rd party CNI plugin that moves
+SR-IOV InfiniBand device to pod namespace and configures device attributes
+such as InfiniBand GUID. It will be delivered in a separate image and copied
+to node CNI directory by a DaemonSet.
+
+#### InfiniBand Kubernetes Daemon
+
+InfiniBand Kubernetes Daemon `ib-kubernetes` is a DaemonSet that watches
+Pod object changes and manages network partitioning via InfiniBand subnet
+manager plugins.
+
+#### SR-IOV Operator
+
+SR-IOV Operator API definition will support `infiniBand` or `ib` device type
+in both `SriovNodeNetworkState` and `SriovNodeNetworkPolicy`.
+
+SR-IOV Operator will support generating net-attach-def Custom Resource with
+`ib-sriov-cni` as CNI plugin type.
+
+#### SR-IOV Config Daemon
+
+SR-IOV Config Daemon, a sub component managed by SR-IOV Operator, will be
+enhanced to change device link type to InfiniBand.
```

Comment:
Note: this will require changes in the mellanox plugin.
Today it only supports configuring link type to ethernet.